site/docs/id/hosting/firebase.md

```diff
@@ -0,0 +1,185 @@
+
+## Persyaratan
+
+Untuk mengikuti panduan ini, kamu perlu memiliki sebuah akun Google.
+Kalau belum punya, kamu bisa membuatnya [di sini](https://accounts.google.com/signup).
+
+## Persiapan
+
+Pada bagian ini, kami akan memandu kamu melalui proses persiapan yang diperlukan.
+Jika kamu memerlukan penjelasan yang lebih detail untuk setiap langkah-langkahnya, silahkan lihat [dokumentasi resmi Firebase](https://firebase.google.com/docs/functions/get-started).
+
+### Membuat Proyek Firebase
+
+1. Buka [Firebase console](https://console.firebase.google.com/) lalu klik **Add Project** untuk menambahkan project baru.
+2. Jika diminta, tinjau dan setujui syarat serta kebijakan Firebase.
+3. Klik **Continue** untuk melanjutkan.
+4. Tentukan apakah kamu ingin menggunakan _analytic_ atau tidak.
+5. Klik **Create Project** untuk membuat project baru.
+
+### Menyiapkan Pengaturan
+
+Untuk membuat sebuah function serta men-deploy-nya ke runtime Firebase Functions, kamu perlu menginstal Node.js dan Firebase CLI terlebih dahulu.
+
+> Perlu diketahui bahwa Firebase Functions hanya mendukung Node.js versi 14, 16, dan 18.
+> Untuk informasi lebih lanjut mengenai versi Node.js yang didukung, lihat [dokumentasi berikut](https://firebase.google.com/docs/functions/manage-functions#set_nodejs_version).
+
+Sesudah Node.js dan NPM terpasang, instal Firebase CLI secara global:
+
+```sh
+npm install -g firebase-tools
+```
+
+### Menyiapkan Proyek
+
+1. Jalankan `firebase login` untuk mengautentikasi Firebase CLI dengan akun kamu melalui browser.
+2. `cd` ke direktori proyek kamu.
+3. Jalankan `firebase init functions`, lalu ketik `y` ketika diberi pertanyaan apakah kamu ingin membuat codebase baru.
+4. Pilih `use existing project`, lalu pilih proyek yang telah kamu buat pada langkah 1.
+5. CLI akan memberikan dua opsi dukungan bahasa pemrograman:
+   - JavaScript
+   - TypeScript
+6. Opsional, pilih opsi ESLint.
+7. CLI kemudian akan bertanya apakah kamu ingin menginstal dependency melalui npm.
+   Tolak jika kamu menggunakan package manager lain, misalnya `yarn` atau `pnpm`.
+   Konsekuensinya, kamu harus `cd` ke direktori `functions` lalu instal dependency secara manual.
+8. Buka file `./functions/package.json` lalu cari key: `"engines": {"node": "16"}`.
+   Versi `node` harus sesuai dengan versi Node.js yang terinstal.
+   Jika tidak sesuai, proyekmu kemungkinan besar tidak akan bisa berjalan.
+
+## Menyiapkan Kode
+
+Kamu bisa menggunakan contoh bot singkat ini sebagai permulaan:
+
+```ts
+import * as functions from "firebase-functions";
+import { Bot, webhookCallback } from "grammy";
+
+const bot = new Bot(""); // <-- taruh token bot-mu di antara ""
+
+bot.command(
+  "start",
+  (ctx) => ctx.reply("Selamat datang! Bot berhasil dijalankan."),
+);
+bot.command("ping", (ctx) => ctx.reply(`Pong! ${new Date()}`));
+
+// Selama mengembangkan bot, kamu bisa men-trigger function kamu melalui https://localhost/<firebase-namaproyek>/us-central1/helloworld
```

Comment:
```
// Selama mengembangkan bot, kamu bisa men-trigger function kamu melalui https://localhost/<firebase-namaproyek>/us-central1/helloworld
```

Is the `helloworld` javascript file?

Reply:
I am not sure. Perhaps @gertminov could elaborate.